core/capabilities/webapi/trigger/trigger.go

```diff
@@ -0,0 +1,299 @@
+package webapi
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sync"
+
+	ethCommon "github.com/ethereum/go-ethereum/common"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/core"
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/api"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/connector"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/handlers/common"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/handlers/webapicapabilities"
+)
+
+const defaultSendChannelBufferSize = 1000
+
+var webapiTriggerInfo = capabilities.MustNewCapabilityInfo(
+	webapicapabilities.TriggerType,
+	capabilities.CapabilityTypeTrigger,
+	"A trigger to start workflow execution from a web api call",
+)
+
+type Input struct {
+}
+type TriggerConfig struct {
+	AllowedSenders []string `toml:"allowedSenders"`
+	AllowedTopics  []string `toml:"allowedTopics"`
+	// RateLimiter    common.RateLimiterConfig `toml:"rateLimiter"`
+	RateLimiter    *values.Map `toml:"rateLimiter"`
+	RequiredParams []string    `toml:"requiredParams"`
+}
+
+type webapiTrigger struct {
+	allowedSendersMap map[string]bool
+	allowedTopicsMap  map[string]bool
+	ch                chan<- capabilities.TriggerResponse
+	config            TriggerConfig
+	rateLimiter       *common.RateLimiter
+}
+
+type triggerConnectorHandler struct {
+	services.StateMachine
+
+	capabilities.CapabilityInfo
+	capabilities.Validator[TriggerConfig, Input, capabilities.TriggerResponse]
+	connector           connector.GatewayConnector
+	lggr                logger.Logger
+	mu                  sync.Mutex
+	registeredWorkflows map[string]webapiTrigger
+}
+
+var _ capabilities.TriggerCapability = (*triggerConnectorHandler)(nil)
+var _ services.Service = &triggerConnectorHandler{}
+
+func NewTrigger(config string, registry core.CapabilitiesRegistry, connector connector.GatewayConnector, lggr logger.Logger) (*triggerConnectorHandler, error) {
+	if connector == nil {
+		return nil, errors.New("missing connector")
+	}
+	handler := &triggerConnectorHandler{
+		Validator:           capabilities.NewValidator[TriggerConfig, Input, capabilities.TriggerResponse](capabilities.ValidatorArgs{Info: webapiTriggerInfo}),
+		connector:           connector,
+		registeredWorkflows: map[string]webapiTrigger{},
+		lggr:                lggr.Named("WorkflowConnectorHandler"),
+	}
+
+	return handler, nil
+}
+
+// Iterate over each topic, checking against senders and rateLimits, then starting event processing and responding
```

Comment:
go doc convention is to start comments with function name
```suggestion
// processTrigger iterates over each topic, checking against senders and rateLimits, then starting event processing and responding
```